client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -99,8 +116,39 @@ private[sharing] case class QueryTableRequest(
   maxFiles: Option[Int],
   pageToken: Option[String],
   includeRefreshToken: Option[Boolean],
-  refreshToken: Option[String]
-)
+  refreshToken: Option[String],
+  idempotencyKey: Option[String],
+  requestedColumns: Seq[String]
+) extends PaginationRequest {
+  override def clone(
```

Comment:
This feels confusing...can you not just set refreshToken = None in the code instead of having this `unsetRefreshToken` parameter?

Reply:
Why don't we get rid of this clone method and directly use copy in the function?